server/routes/monitor.ts

```diff
@@ -151,17 +151,33 @@ export default function routes({ serviceCatalogueService, redisService, dataFilt
     return res.json(environments)
   })
 
+  const getHealthStatus = (json: string): string => {
+    const healthPayload = (json && JSON.parse(json)) || { status: 'UNKNOWN' }
+
+    if (Object.prototype.hasOwnProperty.call(healthPayload, 'status')) {
+      return healthPayload.status
+    }
+    return healthPayload.healthy === true ? 'UP' : 'DOWN'
+  }
+
   post('/queue', async (req, res) => {
-    const streams = Object.keys(req.body?.streams).map(queueName => {
-      return {
-        key: queueName,
-        id: req.body?.streams[queueName],
-      }
-    })
+    const versions = await redisService.readLatest('latest:versions')
+    const health = await redisService.readLatest('latest:health')
+
+    const envNames = Array.from(new Set([health, versions].flatMap(Object.keys)))
+
+    const result = envNames.reduce((acc, key) => {
+      const { v: version } = versions[key] || {}
+      const { json, dateAdded } = health[key] || {}
+
+      const lastMessageTime = relativeTimeFromNow(new Date(dateAdded))
+      const healthStatus = getHealthStatus(json)
 
-    const messages = await redisService.readStream(streams)
+      acc[key] = { version, lastMessageTime, healthStatus }
+      return acc
+    }, {})
 
-    return res.send(messages)
+    return res.send(JSON.stringify(Object.entries(result)))
```

Comment:
guess some of this could be pulled down into the service

Reply:
would make it a bigger change if we wanted to turn it back into  streams and similar to the amount of work the other endpoints are doing.